q2cli/handlers.py

```diff
@@ -104,6 +105,43 @@ def fail(*_):
             return fail
 
 
+class CommandConfigHandler(Handler):
+    """Meta handler which returns a fallback function as its value."""
+
+    def __init__(self, cli_plugin, cli_action):
+        self.cli_plugin = cli_plugin
+        self.cli_action = cli_action
+        super().__init__('cmd_config')
+
+    def get_click_options(self):
+        yield click.Option(
+            ['--' + self.cli_name],
+            type=click.Path(exists=True, dir_okay=False, file_okay=True,
+                            readable=True),
+            help='Use config file for command options')
+
+    def get_value(self, arguments, fallback=None):
+        try:
+            path = self._locate_value(arguments, fallback=fallback)
+            config = configparser.ConfigParser()
+            config.read(path)
+            config_group = config['.'.join([self.cli_plugin, self.cli_action])]
```

Comment:
This will fail with (likely) a `KeyError` if the config file doesn't have a matching section for the plugin command. In that case I think it makes sense to raise a `UserWarning` stating that the config file provided via `--cmd-config` does not have a matching section for the specified plugin command.